### PR TITLE
Add attachment support for v3

### PR DESCRIPTION
**Pull request by lholden**
Adds attachment support for the v3 API. Didn't flesh out disposition and content_id however.

### CODE REVIEW

File: src/v3.rs [hunk 1]
```diff
@@ -140,6 +162,17 @@ impl SGMailV3 {
         self.personalizations.push(p);
     }
 
+    pub fn add_attachment(&mut self, a: Attachment) {
```
Comment (gsquire) [inline src/v3.rs]: Please add a doc comment here as well.

File: Cargo.toml [hunk 2]
```diff
@@ -16,3 +16,4 @@ serde = "0.9"
 serde_derive = "0.9"
 serde_json = "0.9"
 url = "1.2"
+data-encoding = "1.2.0"
```
Comment (lholden) [inline Cargo.toml]: Ah.. I normally do too... didn't mean to pin it so specific :D.

**Comment by lholden** [comment]
Sorry for the messy lack of comments from earlier. It was a busy day... haha. :D

File: src/v3.rs [hunk 3]
```diff
@@ -76,6 +81,25 @@ pub struct Personalization {
     send_at: Option<u64>,
 }
 
+/// An attachment block for a V3 message. Content and filename are required. If the
+/// mime_type is unspecified, the email will use Sendgrids default for attachments
```
Comment (gsquire) [inline src/v3.rs]: Add an apostrophe to "Sendgrid's" and after "attachments" add "which is" so it flows a little better.

Comment (lholden) [inline src/v3.rs]: Alrighty, done. :)

**Comment by gsquire** [comment]
Thanks again for adding this in :)

**Comment by gsquire** [comment]
This will be release under version 0.5.1.

**Comment by lholden** [comment]
Certainly!

Thanks for having a sendgrid crate already so I didn't have to start from scratch. :)

**Pull request merged**